addons/velero/1.6.2/install.sh

```diff
@@ -1,34 +1,63 @@
-
+# shellcheck disable=SC2148
 function velero_pre_init() {
     if [ -z "$VELERO_NAMESPACE" ]; then
         VELERO_NAMESPACE=velero
     fi
     if [ -z "$VELERO_LOCAL_BUCKET" ]; then
         VELERO_LOCAL_BUCKET=velero
     fi
+    # TODO (dans): make this configurable from the installer spec
+    # if [ -z "$VELERO_REQUESTED_CLAIM_SIZE" ]; then
+    #     VELERO_REQUESTED_CLAIM_SIZE="50Gi"
+    # fi
 }
 
+# runs on first install, and on version upgrades only
 function velero() {
     local src="$DIR/addons/velero/$VELERO_VERSION"
     local dst="$DIR/kustomize/velero"
 
-    cp "$src/kustomization.yaml" "$dst/"
+    render_yaml_file "$src/tmpl-kustomization.yaml" > "$dst/kustomization.yaml"
 
     velero_binary
 
+    determine_velero_pvc_size
+
     velero_install "$src" "$dst"
 
     velero_patch_restic_privilege "$src" "$dst"
 
     velero_kotsadm_restore_config "$src" "$dst"
 
+    # always patch the velero and restic manifests to include the PVC case
+    velero_patch_internal_pvc_snapshots "$src" "$dst"
+
     velero_patch_http_proxy "$src" "$dst"
 
     velero_change_storageclass "$src" "$dst"
 
+    velero_migrate_from_object_store "$src" "$dst"
+
     kubectl apply -k "$dst"
 
     kubectl label -n default --overwrite service/kubernetes velero.io/exclude-from-backup=true
+
+    # Bail if the migratoin fails, preventing the original object store from being deleted
+    if [ "$WILL_MIGRATE_VELERO_OBJECT_STORE" = "1" ]; then
```

Comment:
```suggestion
    if [ "$BETA_MIGRATE_VELERO_OBJECT_STORE" = "1" ]; then
```

Reply:
Nope, this is already checked above in `velero_migrate_from_object_store`.

I refactored a bit anyway for clarity.